modules/data/src/main/scala/com.snowplowanalytics.iglu/client/ClientError.scala

```diff
@@ -33,14 +33,17 @@ sealed trait ClientError extends Product with Serializable {
 
 object ClientError {
 
+  val SupersededByField = "supersededBy"
+
   /** Error happened during schema resolution step */
   final case class ResolutionError(value: SortedMap[String, LookupHistory]) extends ClientError {
     def isNotFound: Boolean =
       value.values.flatMap(_.errors).forall(_ == RegistryError.NotFound)
   }
 
   /** Error happened during schema/instance validation step */
-  final case class ValidationError(error: ValidatorError) extends ClientError
+  final case class ValidationError(error: ValidatorError, supersededBy: Option[String])
```

Comment:
There might be a problem here.... I _think_ this case class is used to build some of the bad row types.  And they are described by a schema [e.g. this one](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow.badrows/enrichment_failures/jsonschema/2-0-0#L92).  So we might break bad rows if we add an extra field here.

All of that needs checking on Monday, I might be wrong.

Is there any way we can get get the relevant information in there without changing the bad rows schema?  For example, stuff the supersededBy details into ValidatorError?

Reply:
I've missed that, good spot! It might not be pretty but I guess we can add a message for superseding info to `messages` list of `ValidatorError.InvalidData`. I will push the related commit shortly.

Reply:
Looking at this afresh on Monday.... maybe I was too quick to dismiss this idea to add a field to `ValidationError`.  Sure, it means we need to release a new version of the bad rows schema, but that's probably OK isn't it?  It's a non-breaking change to the schema.

Reply:
Yep, I guess that would be okay. In that case, we would have to update the [schema_violations](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow.badrows/schema_violations/jsonschema/2-0-0) schema as well. Also, we need to release new version of [snowplow-badrows](https://github.com/snowplow/snowplow-badrows/blob/master/src/main/scala/com.snowplowanalytics.snowplow.badrows/Schemas.scala#L34) library with updated schema versions. It look like snowplow-badrows' latest version is on the Cats Effect 3 at the moment. We need to branch out new version from older version that doesn't contains Cats Effect 3.

I also pushed the commit that attaches the superseding info to `ValidatorError`. I know it doesn't look good because we are using the fields in `message` outside of their original purposes but superseding info is attached at least with this way. If you doesn't like this way at all, I can start to look into updating bad rows schema.

Reply:
After a bit more thinking, let's do this proper way with updating bad rows schemas instead of taking short cut. Otherwise, it would be very hard to extract superseding info from bad rows. I will make necessary changes on the bad rows.